### DISCUSSION TITLE
Why HyperPyYAML?

### DISCUSSION THREAD

**Discussion by courentin** [speechbrain/speechbrain#1812] [General]
I see that the speechbrain team has created HyperPyYAML and used it in all the examples.

Given that there's a bunch of hyperparameter tooling that exists in the python community (such as [Hydra](https://github.com/facebookresearch/hydra) for example), it seems to me that HyperPyYAML tried to reinvent the wheel, but I might miss something. Thus, I'm wondering what problems does HyperPyYAML solve that other toolings don't?

**Comment by Gastron** [comment]
There are indeed many different tools that could be used for similar purposes as HyperPyYAML. Hydra is perhaps more heavy-duty, and at the other end even just plain YAML could also be used.

HyperPyYAML came from us starting with plain YAML and finding some key pain points in working with it. In my personal experience, HyperPyYAML's simple syntax for instantiating objects allows for fast work on the kinds of experiments we run in SpeechBrain. Also, the extended references in HyperPyYAML have been very useful for me. Yet HyperPyYAML is still quite light weight and relatively decoupled from the experiment code itself.

Though SpeechBrain recipes use HyperPyYAML (it is also just nice to have a consistent style), if you prefer Hydra, or something else, SpeechBrain does not (and does not want to) impose HyperPyYAML on you.
